modules/client/src/internal/client/methods.ts

```diff
@@ -1116,408 +1101,163 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     return version;
   }
 
-  public isDaoUpdate(
-    actions: DaoAction[],
-  ): boolean {
-    const initializeFromInterface = DAO__factory.createInterface()
-      .getFunction("initializeFrom").format("minimal");
-    return findActionIndex(actions, initializeFromInterface) !== -1;
-  }
-  /**
-   * Check if the specified actions try to update a plugin
-   *
-   * @param {DaoAction[]} actions
-   * @return {*}  {boolean}
-   * @memberof ClientMethods
-   */
-  public isPluginUpdate(
-    actions: DaoAction[],
-  ): boolean {
-    const applyUpdateInterface = PluginSetupProcessor__factory.createInterface()
-      .getFunction("applyUpdate").format("minimal");
-    return findActionIndex(actions, applyUpdateInterface) !== -1;
-  }
-
   /**
-   * check if permission is root
-   * check if permissionId is root
-   * check if where is the dao address
-   * check if who is the psp address
+   * Given a proposal id returns if that proposal is a dao update proposal
    *
-   * @private
-   * @param {(GrantPermissionDecodedParams | RevokePermissionDecodedParams)} params
-   * @param {string} daoAddress
-   * @return {*}  {boolean}
+   * @param {string} proposalId
+   * @return {*}  {Promise<boolean>}
    * @memberof ClientMethods
    */
-  private isPluginUpdatePermissionValid(
-    params: GrantPermissionDecodedParams | RevokePermissionDecodedParams,
-    daoAddress: string,
-  ): boolean {
-    const pspAddress = this.web3.getAddress("pluginSetupProcessorAddress");
-    return (
-      params.permission === Permissions.ROOT_PERMISSION &&
-      params.permissionId === PermissionIds.ROOT_PERMISSION_ID &&
-      params.where === daoAddress &&
-      params.who === pspAddress
-    );
+  public async isDaoUpdate(
```

Comment:
```suggestion
  public async isDaoUpdateProposal(
```

Reply:
Can we put the new methods into a separate file? E.g. called security-check or something related?
The file is already quite long

Reply:
We could move the implementation, not the function, but we will need to pass all the required params because they wont have access to the context and the rest of stuff on `this`